### PR TITLE
[LIVE-9074][LLD][Stax Onboarding] Infinite loading when trying to onboard LNX through Stax onboarding

### PR DESCRIPTION
**Pull request by jdabbech-ledger**
### 📝 Description

Fix an infinite loop on select ledger stax while LNX plugged during onboarding

### ❓ Context

- **Impacted projects**: `LLD` 
- **Linked resource(s)**: [LIVE-9074](https://ledgerhq.atlassian.net/browse/LIVE-9074) 

### ✅ Checklist

- [ ] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo

https://github.com/LedgerHQ/ledger-live/assets/145363160/98374c99-f372-42a4-8c02-5bbfee910cd3

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

[LIVE-9074]: https://ledgerhq.atlassian.net/browse/LIVE-9074?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-desktop/src/renderer/components/SyncOnboarding/DeviceConnection/Searching.tsx [hunk 1]
```diff
@@ -21,6 +23,14 @@ const SyncOnboardingDeviceConnectionSearching = ({
   const history = useHistory();
   const { setDeviceModelId } = useContext(OnboardingContext);
   const theme = useTheme();
+  const currentDevice = useSelector(getCurrentDevice);
+
+  useEffect(() => {
+    if (currentDevice && currentDevice.modelId !== deviceModelId) {
+      setDeviceModelId(currentDevice.modelId);
+      history.replace("/onboarding/select-use-case");
```
Comment (alexandremgo) [inline apps/ledger-live-desktop/src/renderer/components/SyncOnboarding/DeviceConnection/Searching.tsx:31]: On the demo video it does not seem to redirect automatically to the device model selection of the onboarding - the user has to click on the exit button. Is it normal ?

Comment (jdabbech-ledger) [reply apps/ledger-live-desktop/src/renderer/components/SyncOnboarding/DeviceConnection/Searching.tsx:31]: During the demo I clicked on the exit button in order to skip the install apps step during onboarding. 
It's not related to this PR and the `history.replace` is used here to replace the stax onboarding page by the LNX one

Comment (alexandremgo) [reply apps/ledger-live-desktop/src/renderer/components/SyncOnboarding/DeviceConnection/Searching.tsx:31]: Misunderstanding on the demo video hehe

**Review by alexandremgo — APPROVED** [review]
LGTM

**Comment by jdabbech-ledger** [comment]
The demo scenario is:
- click on Stax onboarding with LNX plugged -> redirection to LNX onboarding
- go back
- click on Stax onboarding with Stax plugged -> continue with Stax onboarding

**Pull request merged**